### ISSUE TITLE
special characters in domain name not working

### ISSUE THREAD

**Issue by whitemacs** [gophish/gophish#2962]
What version of Gophish are you using?:
GoPhish v.0.12.1

Brief description of the issue: 
When using a domain with a special character, such as "í", sending e-mails is not working.
If I send a test mail, the message:

" Max connection attempts exceeded - 535 Authentication credentials invalid"

appears, even though the credentials are 100% correct. My humble assumption is, that the dialer is not working with special characters.
I made a debug output just before 

```
d := gomail.NewWithDialer(dialer, host, port, s.Username, s.Password)
```

in the smtp.go file, and there all information are stored correctly. Afterwards my abilities aren't strong enough to debug further...

To circumvent the smtp validation, I used the following answer in an other issue:
https://github.com/gophish/gophish/issues/2783#issuecomment-1500974082

What are you expecting to see happen? :
Test e-mails sending out

What are you seeing happen?

Please provide any terminal output that may be relevant below:

```
time="2023-08-31T07:43:33Z" level=info msg="info@xxxí.com"
time="2023-08-31T07:43:33Z" level=warning msg="Max connection attempts exceeded - 535 Authentication credentials invalid"
time="2023-08-31T07:43:33Z" level=error msg="Max connection attempts exceeded - 535 Authentication credentials invalid"

```

**Comment by whitemacs** [comment]
Hey @glennzw,

I am amazed by your reply, thank you for taking your time!!

Unfortunately I am currently on vacation till next Friday. As soon as I will come back, I will try to reproduce what you have done and will reply accordingly.

Thank you once more!

**Comment by glennzw** [comment]
Sure no worries, enjoy the vacation and we can catch up once you're back!

**Comment by whitemacs** [comment]
Hello again, @glennzw 

I followed your steps and can indeed confirm, that the correct information was send out to the mail server.

After I found out that this was the case, I asked my provider (IONOS), it seems that they had to enable something first.
It  still doesn't work with the special í directly, but via punycode, which is fine for me.

Thank you for your support and sorry that I opened an issue even though it wasn't a gophish issue afterall.
